lib/ansible/modules/cloud/amazon/elb_target.py

```diff
@@ -0,0 +1,245 @@
+description:
+    - Used to register or deregister a target in a target group
+version_added: "2.4"
+author: "Rob White (@wimnat)"
+options:
+  target_group_arn:
+    description:
+      - The Amazon Resource Name (ARN) of the target group. Mutually exclusive of I(target_group_name).
+  target_group_name:
+    description:
+      - The name of the target group. Mutually exclusive of I(target_group_arn).
+  target_id:
+    description:
+      - The ID of the target.
+    required: true
+  target_port:
+    description:
+      - The port on which the target is listening. You can specify a port override. If a target is already registered,
+        you can register it again using a different port.
+    required: false
+    default: The default port for a target is the port for the target group.
+  state:
+    description:
+      - Register or deregister the target.
+    required: true
+    choices: [ 'present', 'absent' ]
+extends_documentation_fragment:
+    - aws
+    - ec2
+notes:
+  - If you specified a port override when you registered a target, you must specify both the target ID and the port when you deregister it.
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Register a target to a target group
+- elb_target:
+    target_group_name: mytargetgroup
+    target_id: i-1234567
+    state: present
+
+# Deregister a target from a target group
+- elb_target:
+    target_group_name: mytargetgroup
+    target_id: i-1234567
+    state: absent
+
+# Modify a target to use a different port
+# Register a target to a target group
+- elb_target:
+    target_group_name: mytargetgroup
+    target_id: i-1234567
+    target_port: 8080
+    state: present
+
+'''
+
+RETURN = '''
+
+'''
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (boto3_conn, camel_dict_to_snake_dict,
+                                      ec2_argument_spec, get_aws_connection_info)
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def convert_tg_name_to_arn(connection, module, tg_name):
+
+    try:
+        response = connection.describe_target_groups(Names=[tg_name])
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    tg_arn = response['TargetGroups'][0]['TargetGroupArn']
+
+    return tg_arn
+
+
+def describe_targets(connection, module, tg_arn, target):
+
+    """
+    Describe targets in a target group
+
+    :param module: ansible module object
+    :param connection: boto3 connection
+    :param tg_arn: target group arn
+    :param target: dictionary containing target id and port
+    :return:
+    """
+
+    try:
+        return connection.describe_target_health(TargetGroupArn=tg_arn, Targets=target)['TargetHealthDescriptions'][0]
```

Comment:
Also, what if the target has been registered on multiple ports (since running the playbook with a new port doesn't deregister the target with the old port)? Should those be in the returned info as well, instead of just the first item in the list?